xtask/src/tempdir.rs

```diff
@@ -0,0 +1,61 @@
+use std::{
+    env,
+    ffi::{OsStr, OsString},
+    fs::{self, create_dir},
+    io,
+    path::{Path, PathBuf},
+};
+use uuid::Uuid;
+
+/// A temporary directory which is cleaned on `drop` (unless the `preserve` field
+/// is `true`.
+pub struct TempDir {
+    /// Path to the build directory.
+    dir_path: PathBuf,
+    /// Whether to preserve the build directory after `drop`. If `false`, it
+    /// gets removed automatically.
+    preserve: bool,
+}
+
+impl TempDir {
+    pub fn new(prefix: &str, preserve: bool) -> Self {
+        let dir_path = env::temp_dir().join(format!("{prefix}-{}", Uuid::new_v4()));
+        Self { dir_path, preserve }
+    }
+
+    pub fn create(&self) -> io::Result<()> {
+        create_dir(self.dir_path.as_path())
+    }
```

Comment:
please move the remove the `create` and move the `create_dir` inside the `new` to have a more idiomatic rust constructor